### ISSUE TITLE
os_patching_fact_generation.sh Script runs in nil:NilClass (NoMethodError)

### ISSUE THREAD

**Issue by epgkue** [albatrossflavour/puppet_os_patching#226]
## Affected Puppet, Ruby, OS and module versions/distributions
- Puppet: 7.23.0
- Ruby: ruby 2.7.7p221 AND ruby 3.0.4p208
- Distribution: AlmaLinux release 8.7 (Stone Smilodon) AND AlmaLinux release 9.1 (Lime Lynx)
- Module version: albatrossflavour-os_patching (v0.19.0)

## How to reproduce (e.g Puppet code you use)
1. Assign os_patching class to host
2. Run puppet agent
3. Running bash in debug script locally:
 bash -x /usr/local/bin/os_patching_fact_generation.sh

## What are you seeing
....
++ /opt/puppetlabs/puppet/bin/ruby -e 'require '\''json'\''; json_hash = JSON.parse(ARGF.read); json_hash['\''resources'\''].select { |r| r['\''type'\''] == '\''Package'\'' and r['\''parameters'\'']['\''ensure'\''].match /\d.+/ }.each do | m | puts m['\''title'\''] end'
Traceback (most recent call last):
        2: from -e:1:in `<main>'
        1: from -e:1:in `select'
-e:1:in `block in <main>': undefined method `match' for nil:NilClass (NoMethodError)
....

## What behaviour did you expect instead
Updating os_patching facts to the Hosts to Foreman.

**Comment by albatrossflavour** [comment]
I've not tested os_patching on alma yet, but sounds like that could be the culprit.  We've also seen some issues using open source puppet with `os_patching` (I was designed to work with PE).  I'll try and run up a test environment and see what I can find.

**Comment by albatrossflavour** [comment]
I've not been able to replicate this:
![CleanShot 2023-07-11 at 21 33 17](https://github.com/albatrossflavour/puppet_os_patching/assets/2277476/ff88bd3b-24b9-4b8d-b79f-3e01bf3eee5c)

I've tried a few different combos, but no issues.

I did the following with a bare Alma instance:

```
rpm -Uvh https://yum.puppet.com/puppet7-release-el-8.noarch.rpm
yum install puppet-agent
puppet module install albatrossflavour/os_patching
puppet apply -e 'include os_patching'
bash -x /usr/local/bin/os_patching_fact_generation.sh
```

Did I miss anything?

**Comment by albatrossflavour** [comment]
@epgkue do you have any other steps I can try to reproduce this?

**Comment by dirkonet** [comment]
At least on my CentOS 7 machines, this seems to be the same issue as seen in #202 - one of my packages does not have an "ensure" parameter in the catalog, causing the array access to fail.

While this was fixed with #203, the patch seems to have been overwritten and is not in the current development branch or release. I'd suggest to re-apply #203.

**Comment by albatrossflavour** [comment]
Well spotted @dirkonet .  I've applied the fix to development and will release an update soon.
